websites/views.py

```diff
@@ -342,22 +356,22 @@ class WebsiteContentViewSet(
 
     def get_queryset(self):
         parent_lookup_website = self.kwargs.get("parent_lookup_website")
-        filter_type_keys = [
-            qs_key
-            for qs_key in self.request.query_params.keys()
-            # View should accept "type" as a single query param, or multiple types,
-            # e.g.: "?type[0]=sometype&type[1]=othertype"
-            if qs_key == "type" or qs_key.startswith("type[")
-        ]
-
-        queryset = WebsiteContent.objects.filter(website__name=parent_lookup_website)
-        if filter_type_keys:
-            queryset = queryset.filter(
-                type__in=[
-                    self.request.query_params[filter_type_key]
-                    for filter_type_key in filter_type_keys
-                ]
-            )
+        search = self.request.query_params.get("search")
+        types = _get_value_list_from_query_params(self.request.query_params, "type")
+        text_ids = _get_value_list_from_query_params(
+            self.request.query_params, "text_id"
+        )
+
+        queryset = WebsiteContent.objects.filter(
+            website__name=parent_lookup_website
+        ).select_related("website", "website__starter")
+        if types:
+            queryset = queryset.filter(type__in=types)
+        if search:
```

Comment:
Can we add tests for `search` and `text_ids` having a value?

Reply:
I already added tests for these cases: https://github.com/mitodl/ocw-studio/blob/gs/async-search-backend/websites/views_test.py#L473